airflow/api_fastapi/core_api/routes/public/dags.py

```diff
@@ -303,3 +307,45 @@ def delete_dag(
             status.HTTP_409_CONFLICT, f"Task instances of dag with id: '{dag_id}' are still running"
         )
     return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@dags_router.get(
+    "/{dag_id}/assets/queuedEvent",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_401_UNAUTHORIZED,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+def get_dag_asset_queued_events(
+    dag_id: str,
+    session: Annotated[Session, Depends(get_session)],
+    before: str = Query(None),
```

Comment:
I thought of it earlier, the problem with that is that the `_safe_parse_datetime` expects a value which kind of makes the "before" a mandatory field which it is not. I couldn't see an easy way of extending the `_safe_parse_datetime` correctly. It also didnt feel right to do that because we were checking for mandatory field.

Let me try if somethign like: `Optional[DateTimeQuery] = None` works.

Reply:
Yep, it works fine.

![image](https://github.com/user-attachments/assets/902275b3-7d54-4b1f-9384-a4cadfa5320f)

![image](https://github.com/user-attachments/assets/c8274bbc-3c35-492e-ac67-9d3baadf8d9c)

Reply:
You mentionned using `Optional[DateTimeQuery] = None` but the code stills mention `before: str = Query(None)` is that normal?

Reply:
Fixed with a new OptionalDateTimeQuery